src/sparseml/modifiers/quantization/utils/quantize.py

```diff
@@ -231,11 +231,15 @@ def add_output_activation_observers(module: Module):
     """
     # adapted from torch/ao/quantization/quantize.py::_add_observer_
     # source: https://github.com/pytorch/pytorch/blob/v1.13.0/torch/ao/quantization/quantize.py#L135  # noqa: E501
-    try:
-        device = next(module.parameters()).device
-    except StopIteration:
-        # default to CPU if module has no parameters
-        device = "cpu"
+
+    def get_device(m):
+        # infers device based on either paramteres or buffers
+        unique_devices = {p.device for p in m.parameters()} | {
+            p.device for p in m.buffers()
+        }
+
+        assert len(unique_devices) <= 1
```

Comment:
need to raise a specific error here - generally we try to avoid asserts that users could hit

Reply:
Sure, will update the PR.